### PR TITLE
Commonly used function needed for NooBaa Standalone deployment

### PR DESCRIPTION
**Pull request by vavuthu**
Signed-off-by: vavuthu [vavuthu@redhat.com](mailto:vavuthu@redhat.com)

### CODE REVIEW

**Comment by vavuthu** [comment]
verifed the all the function with deployment

```
$ noobaa-sa-install --conf packages.yaml 
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.8.16, pytest-7.4.2, pluggy-1.3.0 -- /home/redhat/venv_infra1/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/redhat/noobaa-sa-infra
configfile: pytest.ini
collected 1 item                                                                                                                                                                                           

tests/deployment/test_deployment.py::test_deployment 
---------------------------------------------------------------------------------------------- live log call -----------------------------------------------------------------------------------------------
2023-10-09 09:43:58 - MainThread - deployment.deployment - INFO - Installing Noobaa Standalone
2023-10-09 09:44:01 - MainThread - common.package_fetcher - INFO - Downloaded RPM package to /tmp/noobaa-core-5.14.0-1-xs3select.el8.x86_64_2f126aql.rpm
2023-10-09 09:44:01 - MainThread - common.rpm_manager - INFO - Installing /tmp/noobaa-core-5.14.0-1-xs3select.el8.x86_64_2f126aql.rpm
2023-10-09 09:44:01 - MainThread - common.command_runner - INFO - Executing command: sudo yum install /tmp/noobaa-core-5.14.0-1-xs3select.el8.x86_64_2f126aql.rpm -y
2023-10-09 09:44:03 - MainThread - common.command_runner - INFO - Command stdout: Last metadata expiration check: 0:07:15 ago on Mon 09 Oct 2023 09:36:46 AM EDT.
Package noobaa-core-5.14.0-1.el8.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!

2023-10-09 09:44:03 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:03 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:03 - MainThread - common.rpm_manager - INFO - Successfully installed /tmp/noobaa-core-5.14.0-1-xs3select.el8.x86_64_2f126aql.rpm
2023-10-09 09:44:03 - MainThread - common.rpm_manager - INFO - Installing https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm
2023-10-09 09:44:03 - MainThread - common.command_runner - INFO - Executing command: sudo yum install https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm -y
2023-10-09 09:44:06 - MainThread - common.command_runner - INFO - Command stdout: Last metadata expiration check: 0:07:18 ago on Mon 09 Oct 2023 09:36:46 AM EDT.
pgdg-redhat-repo-latest.noarch.rpm               27 kB/s |  13 kB     00:00    
Package pgdg-redhat-repo-42.0-35PGDG.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!

2023-10-09 09:44:06 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:06 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:06 - MainThread - common.rpm_manager - INFO - Successfully installed https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm
2023-10-09 09:44:06 - MainThread - common.postgres_utils - INFO - Enable the module stream for Postgres version to 12
2023-10-09 09:44:06 - MainThread - common.command_runner - INFO - Executing command: sudo dnf module enable postgresql:12
2023-10-09 09:44:08 - MainThread - common.command_runner - INFO - Command stdout: Last metadata expiration check: 0:07:20 ago on Mon 09 Oct 2023 09:36:46 AM EDT.
Dependencies resolved.
Nothing to do.
Complete!

2023-10-09 09:44:08 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:08 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:08 - MainThread - common.rpm_manager - INFO - Installing libpq postgresql postgresql-server
2023-10-09 09:44:08 - MainThread - common.command_runner - INFO - Executing command: sudo yum install libpq postgresql postgresql-server -y
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command stdout: Last metadata expiration check: 0:07:23 ago on Mon 09 Oct 2023 09:36:46 AM EDT.
Package libpq-13.5-1.el8.x86_64 is already installed.
Package postgresql-12.12-1.module_el8.6.0+1219+2bd8590a.x86_64 is already installed.
Package postgresql-server-12.12-1.module_el8.6.0+1219+2bd8590a.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!

2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:10 - MainThread - common.rpm_manager - INFO - Successfully installed None
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Executing command: sudo chmod 777 /usr/local/noobaa-core
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command stdout is empty
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:10 - MainThread - common.file_system_utils - INFO - Permissions for '/usr/local/noobaa-core' have been set to 777
2023-10-09 09:44:10 - MainThread - common.file_system_utils - INFO - Directory '/usr/local/noobaa-core/storage' already exists.
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Executing command: sudo chmod 777 /var/run/postgresql
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command stdout is empty
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:10 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:10 - MainThread - common.file_system_utils - INFO - Permissions for '/var/run/postgresql' have been set to 777
2023-10-09 09:44:10 - MainThread - deployment.deployment - INFO - Initializing DB
2023-10-09 09:44:10 - MainThread - deployment.npm - INFO - executing 'npm run db:init'

> noobaa-core@5.14.0 db:init
> initdb -D ./storage/postgres -E UTF-8 --locale=C

The files belonging to this database system will be owned by user "redhat".
This user must also own the server process.

The database cluster will be initialized with locale "C".
The default text search configuration will be set to "english".

Data page checksums are disabled.

initdb: error: directory "./storage/postgres" exists but is not empty
If you want to create a new database system, either remove or empty
the directory "./storage/postgres" or run initdb
with an argument other than "./storage/postgres".
2023-10-09 09:44:11 - MainThread - deployment.deployment - INFO - starting the DB
2023-10-09 09:44:11 - MainThread - deployment.npm - INFO - executing 'npm run db'
2023-10-09 09:44:21 - MainThread - deployment.deployment - INFO - Creating DB, users and permissions
2023-10-09 09:44:21 - MainThread - deployment.npm - INFO - executing 'npm run db:create'

> noobaa-core@5.14.0 db:create
> npm run db:createuser && npm run db:createdb


> noobaa-core@5.14.0 db:createuser
> createuser postgres && psql postgres -c 'ALTER USER postgres CREATEDB'

createuser: error: creation of new role failed: ERROR:  role "postgres" already exists
2023-10-09 09:44:32 - MainThread - deployment.deployment - INFO - creating .env file
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Executing command: sudo touch /usr/local/noobaa-core/.env
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command stdout is empty
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Executing command: sudo chmod 666 /usr/local/noobaa-core/.env
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command stdout is empty
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:32 - MainThread - deployment.deployment - INFO - creating config-local.js file
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Executing command: sudo cp /home/redhat/noobaa-sa-infra/templates/config-local.js /usr/local/noobaa-core/config-local.js
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command stdout is empty
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command stderr is empty
2023-10-09 09:44:32 - MainThread - common.command_runner - INFO - Command return code: 0
2023-10-09 09:44:32 - MainThread - deployment.deployment - INFO - starting the web service
2023-10-09 09:44:32 - MainThread - deployment.npm - INFO - executing 'npm run web'
2023-10-09 09:44:42 - MainThread - deployment.deployment - INFO - starting the bg service
2023-10-09 09:44:42 - MainThread - deployment.npm - INFO - executing 'npm run bg'
2023-10-09 09:44:52 - MainThread - deployment.deployment - INFO - starting the hosted agent service
2023-10-09 09:44:52 - MainThread - deployment.npm - INFO - executing 'npm run hosted_agents'
2023-10-09 09:45:02 - MainThread - deployment.deployment - INFO - starting the s3 endpoint service
2023-10-09 09:45:02 - MainThread - deployment.npm - INFO - executing 'npm run s3'
2023-10-09 09:45:12 - MainThread - common.file_system_utils - INFO - Directory '/usr/local/noobaa-core/storage/backingstores/drive0' already exists.
2023-10-09 09:45:12 - MainThread - deployment.deployment - INFO - running backing store '/usr/local/noobaa-core/storage/backingstores/drive0' at port 9991
2023-10-09 09:45:12 - MainThread - deployment.npm - INFO - executing 'npm run backingstore'
2023-10-09 09:45:22 - MainThread - common.file_system_utils - INFO - Directory '/usr/local/noobaa-core/storage/backingstores/drive1' already exists.
2023-10-09 09:45:22 - MainThread - deployment.deployment - INFO - running backing store '/usr/local/noobaa-core/storage/backingstores/drive1' at port 9992
2023-10-09 09:45:22 - MainThread - deployment.npm - INFO - executing 'npm run backingstore'
2023-10-09 09:45:32 - MainThread - common.file_system_utils - INFO - Directory '/usr/local/noobaa-core/storage/backingstores/drive2' already exists.
2023-10-09 09:45:32 - MainThread - deployment.deployment - INFO - running backing store '/usr/local/noobaa-core/storage/backingstores/drive2' at port 9993
2023-10-09 09:45:32 - MainThread - deployment.npm - INFO - executing 'npm run backingstore'
2023-10-09 09:45:42 - MainThread - common.file_system_utils - INFO - Directory '/usr/local/noobaa-core/storage/backingstores/drive3' already exists.
2023-10-09 09:45:42 - MainThread - deployment.deployment - INFO - running backing store '/usr/local/noobaa-core/storage/backingstores/drive3' at port 9994
2023-10-09 09:45:42 - MainThread - deployment.npm - INFO - executing 'npm run backingstore'
2023-10-09 09:45:52 - MainThread - deployment.deployment - INFO - checking storage status
2023-10-09 09:45:52 - MainThread - deployment.npm - INFO - executing 'npm run api'

> noobaa-core@5.14.0 api
> node --unhandled-rejections=warn src/cmd/api node sync_monitor_to_store

load_config_local: LOADED { DEFAULT_POOL_TYPE: 'HOSTS', AGENT_RPC_PORT: '9999', AGENT_RPC_PROTOCOL: 'tcp', ENDPOINT_HTTP_SERVER_REQUEST_TIMEOUT: 900000, ENDPOINT_HTTP_SERVER_KEEPALIVE_TIMEOUT: 60000, BUCKET_AUTOCONF_TIER2_ENABLED: true, BLOCK_STORE_FS_TMFS_ENABLED: true, BLOCK_STORE_FS_MAPPING_INFO_ENABLED: true, DEDUP_ENABLED: false, IO_CALC_MD5_ENABLED: false, IO_CALC_SHA256_ENABLED: false, MAX_OBJECT_PART_SIZE: 1073741824, IO_CHUNK_READ_CACHE_SIZE: 1073741824, IO_READ_BLOCK_TIMEOUT: 600000, IO_WRITE_BLOCK_TIMEOUT: 600000, IO_DELETE_BLOCK_TIMEOUT: 600000, IO_REPLICATE_BLOCK_TIMEOUT: 600000, NODE_IO_DETENTION_DISABLE: true, NODE_IO_DETENTION_THRESHOLD: 0, CHUNK_SPLIT_AVG_CHUNK: 268435456, CHUNK_SPLIT_DELTA_CHUNK: 0, CHUNK_CODER_DIGEST_TYPE: 'none', CHUNK_CODER_FRAG_DIGEST_TYPE: 'none', CHUNK_CODER_COMPRESS_TYPE: 'none', CHUNK_CODER_CIPHER_TYPE: 'none', CHUNK_CODER_REPLICAS: 1, CHUNK_CODER_EC_DATA_FRAGS: 2, CHUNK_CODER_EC_PARITY_FRAGS: 2, CHUNK_CODER_EC_PARITY_TYPE: 'cm256', CHUNK_CODER_EC_TOLERANCE_THRESHOLD: 2, CHUNK_CODER_EC_IS_DEFAULT: true, SCRUBBER_ENABLED: false, REBUILD_NODE_ENABLED: false, AWS_METERING_ENABLED: false, AGENT_BLOCKS_VERIFIER_ENABLED: false, TIERING_TTL_WORKER_ENABLED: true }
undefined
2023-10-09 09:45:54 - MainThread - deployment.deployment - INFO - Storage Check successfully passed
2023-10-09 09:45:54 - MainThread - deployment.deployment - INFO - checking aggregate node status
2023-10-09 09:45:54 - MainThread - deployment.npm - INFO - executing 'npm run api'

> noobaa-core@5.14.0 api
> node --unhandled-rejections=warn src/cmd/api node aggregate_nodes {}

load_config_local: LOADED { DEFAULT_POOL_TYPE: 'HOSTS', AGENT_RPC_PORT: '9999', AGENT_RPC_PROTOCOL: 'tcp', ENDPOINT_HTTP_SERVER_REQUEST_TIMEOUT: 900000, ENDPOINT_HTTP_SERVER_KEEPALIVE_TIMEOUT: 60000, BUCKET_AUTOCONF_TIER2_ENABLED: true, BLOCK_STORE_FS_TMFS_ENABLED: true, BLOCK_STORE_FS_MAPPING_INFO_ENABLED: true, DEDUP_ENABLED: false, IO_CALC_MD5_ENABLED: false, IO_CALC_SHA256_ENABLED: false, MAX_OBJECT_PART_SIZE: 1073741824, IO_CHUNK_READ_CACHE_SIZE: 1073741824, IO_READ_BLOCK_TIMEOUT: 600000, IO_WRITE_BLOCK_TIMEOUT: 600000, IO_DELETE_BLOCK_TIMEOUT: 600000, IO_REPLICATE_BLOCK_TIMEOUT: 600000, NODE_IO_DETENTION_DISABLE: true, NODE_IO_DETENTION_THRESHOLD: 0, CHUNK_SPLIT_AVG_CHUNK: 268435456, CHUNK_SPLIT_DELTA_CHUNK: 0, CHUNK_CODER_DIGEST_TYPE: 'none', CHUNK_CODER_FRAG_DIGEST_TYPE: 'none', CHUNK_CODER_COMPRESS_TYPE: 'none', CHUNK_CODER_CIPHER_TYPE: 'none', CHUNK_CODER_REPLICAS: 1, CHUNK_CODER_EC_DATA_FRAGS: 2, CHUNK_CODER_EC_PARITY_FRAGS: 2, CHUNK_CODER_EC_PARITY_TYPE: 'cm256', CHUNK_CODER_EC_TOLERANCE_THRESHOLD: 2, CHUNK_CODER_EC_IS_DEFAULT: true, SCRUBBER_ENABLED: false, REBUILD_NODE_ENABLED: false, AWS_METERING_ENABLED: false, AGENT_BLOCKS_VERIFIER_ENABLED: false, TIERING_TTL_WORKER_ENABLED: true }
{
  nodes: { count: 4, online: 4, by_mode: { OPTIMAL: 4 } },
  storage_nodes: { count: 4, online: 4, by_mode: { OPTIMAL: 4 } },
  storage: {
    total: 300500910080,
    free: 280327110656,
    used: 0,
    reserved: 419430400,
    unavailable_free: 0,
    unavailable_used: 0,
    used_other: 19754369024
  },
  data_activities: [ [length]: 0 ]
}
2023-10-09 09:45:56 - MainThread - deployment.deployment - INFO - Aggregate node status check successfully passed
PASSED

====================================================================================== 1 passed in 118.59s (0:01:58) ======================================================================================
```

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+*~
+.*.sw[nmop]
+*.pyc
+.tox
+__pycache__
+ven*
```
Comment (petr-balogh) [inline .gitignore:6]: can we also add .`venv` please?

**Review by petr-balogh — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (vavuthu) [reply .gitignore:6]: done.

File: common/command_runner.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        log.info(f"Command stdout: {completed_process.stdout.decode()}")
+    else:
+        log.info("Command stdout is empty")
+
+    if len(completed_process.stderr) > 0:
+        if not silent:
+            log.info(f"Command stderr: {completed_process.stderr.decode()}")
+    else:
+        log.info("Command stderr is empty")
+    log.info(f"Command return code: {completed_process.returncode}")
```
Comment (dahorak) [inline common/command_runner.py]: I would revisit the log level of those messages - I would propose to use debug.

**Review by dahorak — APPROVED** [review]
I would revisit the usage of log level _info_, otherwise LGTM.

File: common/command_runner.py (hunk 2 above)
Comment (vavuthu) [reply common/command_runner.py]: done in the latest commit

**Review by dahorak — APPROVED** [review]

**Pull request merged**